modules/extension/wmts/src/main/java/org/geotools/ows/wmts/map/WMTSCoverageReader.java

```diff
@@ -549,21 +491,25 @@ public String getMetadataValue(String name) {
     }
 
     /** @return the mapRequest */
+    @Deprecated
     public GetTileRequest getTileRequest() {
-        return tileRequest;
+        throw new UnsupportedOperationException();
```

Comment:
See other comments about deprecation, this is not how deprecation works, you might as well just remove the method. Keeping the method would defeat the purpose of this refactor though... maybe ask on the list about having a direct method removal, without deprecation cycle? It does not seem these methods are used in GT or GS, but still, they are public and usable by other clients....

Reply:
I've kept the functionality, but added a deprecation.